plugin/src/main/java/org/opensearch/ml/action/agents/TransportSearchAgentAction.java

```diff
@@ -42,6 +45,28 @@ private void search(SearchRequest request, ActionListener<SearchResponse> action
         ActionListener<SearchResponse> listener = wrapRestActionListener(actionListener, "Fail to search agent");
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
             ActionListener<SearchResponse> wrappedListener = ActionListener.runBefore(listener, context::restore);
+            // Check if the original query is not null before adding it to the must clause
+            BoolQueryBuilder queryBuilder = QueryBuilders.boolQuery();
+            if (request.source().query() != null) {
+                queryBuilder.must(request.source().query());
+            }
+
+            // Create a BoolQueryBuilder for the should clauses
+            BoolQueryBuilder shouldQuery = QueryBuilders.boolQuery();
+
+            // Add a should clause to include documents where IS_HIDDEN_FIELD is false
+            shouldQuery.should(QueryBuilders.termQuery(MLAgent.IS_HIDDEN_FIELD, false));
```

Comment:
So for search agent API, we won't return hidden Agent. For get Agent API, user can get hidden agent with removing hidden model id ?

Reply:
I think according to arjun's suggestion, I'm going to block the hidden agent get request.